docs/02.API-REFERENCE.md

```diff
@@ -169,6 +169,51 @@ typedef struct
 } jerry_property_descriptor_t;
 ```
 
+## jerry_heap_stats_t
+
+**summary**
+
+
+Description of JerryScript heap memory stats.
+It is for memory profiling.
+
+**Prototype**
+
+```c
+typedef struct
+{
+  size_t size; /**< heap total size */
+
+  size_t allocated_bytes; /**< currently allocated bytes */
+  size_t peak_allocated_bytes; /**< peak allocated bytes */
+
+  size_t waste_bytes; /**< bytes waste due to blocks filled partially */
+  size_t peak_waste_bytes; /**< peak wasted bytes */
+
+  size_t byte_code_bytes; /**< allocated memory for byte code */
+  size_t peak_byte_code_bytes; /**< peak allocated memory for byte code */
+
+  size_t string_bytes; /**< allocated memory for strings */
+  size_t peak_string_bytes; /**< peak allocated memory for strings */
+
+  size_t object_bytes; /**< allocated memory for objects */
+  size_t peak_object_bytes; /**< peak allocated memory for objects */
+
+  size_t property_bytes; /**< allocated memory for properties */
+  size_t peak_property_bytes; /**< peak allocated memory for properties */
+
+  size_t skip_count; /**< Number of skip-aheads during insertion of free block */
+  size_t nonskip_count; /**< Number of times we could not skip ahead during
+                         *   free block insertion */
+
+  size_t alloc_count; /**< number of memory allocations */
+  size_t free_count; /**< number of memory frees */
+  size_t alloc_iter_count; /**< Number of iterations required for allocations */
+  size_t free_iter_count; /**< Number of iterations required for inserting free blocks */
```

Comment:
I would really not make this structure public. It contains implementation details which might change.